examples/cpp/waku.cpp

```diff
@@ -0,0 +1,336 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <argp.h>
+#include <signal.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdint.h>
+#include <vector>
+#include <iostream>
+
+#include <sys/types.h>
+#include <unistd.h>
+#include <sys/syscall.h>
+
+#include "../../library/libwaku.h"
+
+#define WAKU_CALL(call)                                                        \
```

Comment:
If it can be done to outcast the return value to the caller would be nice.
I'm thinking of a templatized function wrapper, but definitely that is more complicated.